### PR TITLE
nrf5x_common: Add qdec peripheral implementation

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR adds a simple implementation of the qdec peripheral for the nrf5x devices.

### Testing procedure

As far as I know there is no board with both an nrf5x chip and a rotary encoder present. 

I have some test output from my own board, but I can imagine that it doesn't look that convincing :)

<details><summary>Output</summary>

```
/home/koen/dev/RIOT-dev/dist/tools/jlink/jlink.sh term-rtt 
### Starting RTT terminal ###
Twisted not available, please install it if you want to use pyterm's JSON capabilities
2021-11-15 15:15:03,208 # Host name for TCP connection is missing, defaulting to "localhost"
2021-11-15 15:15:03,208 # Connect to localhost:19021
Welcome to pyterm!
Type '/exit' to exit.
2021-11-15 15:15:04,211 # SEGGER J-Link V7.58 - Real time terminal output
2021-11-15 15:15:04,212 # J-Link EDU Mini V1 compiled Nov  2 2021 11:12:01 V1.0, SN=801025014
2021-11-15 15:15:04,212 # Process: JLinkExe
2021-11-15 15:15:04,212 # main(): This is RIOT! (Version: 2022.01-devel-503-g9df79-pr/benchmark/ztimer)
2021-11-15 15:15:04,213 # Welcome into Quadrature Decoder (QDEC) test program.
2021-11-15 15:15:04,213 # This program will count pulses on all available QDEC channels
2021-11-15 15:15:04,213 # Written for nucleo-f401re, you have to plug signals A and B as follow :
2021-11-15 15:15:04,213 #   QDEC0 : signal A on PA6 and signal B on PA7
2021-11-15 15:15:04,213 #   QDEC1 : signal A on PB6 and signal B on PB7
2021-11-15 15:15:04,214 # Quadrature decoding mode is set to X4 : counting on all edges on both signals
2021-11-15 15:15:04,214 # QDEC 0 = 0
2021-11-15 15:15:04,214 # QDEC 0 = 0
2021-11-15 15:15:04,214 # QDEC 0 = 0
2021-11-15 15:15:04,214 # QDEC 0 = 0
2021-11-15 15:15:04,214 # QDEC 0 = 0
2021-11-15 15:15:04,214 # QDEC 0 = 0
2021-11-15 15:15:05,128 # QDEC 0 = 0
2021-11-15 15:15:06,128 # QDEC 0 = -3
2021-11-15 15:15:07,127 # QDEC 0 = 0
2021-11-15 15:15:08,117 # QDEC 0 = -1
2021-11-15 15:15:09,129 # QDEC 0 = -2
2021-11-15 15:15:10,116 # QDEC 0 = -6
2021-11-15 15:15:11,117 # QDEC 0 = -6
2021-11-15 15:15:12,117 # QDEC 0 = -28
2021-11-15 15:15:13,117 # QDEC 0 = 0
2021-11-15 15:15:14,108 # QDEC 0 = 0
2021-11-15 15:15:15,108 # QDEC 0 = 2
2021-11-15 15:15:16,108 # QDEC 0 = 0
2021-11-15 15:15:17,110 # QDEC 0 = 2
2021-11-15 15:15:18,109 # QDEC 0 = 0
2021-11-15 15:15:19,099 # QDEC 0 = 2
2021-11-15 15:15:20,103 # QDEC 0 = 0
2021-11-15 15:15:21,110 # QDEC 0 = 0
2021-11-15 15:15:22,104 # QDEC 0 = 2
2021-11-15 15:15:23,098 # QDEC 0 = 0
2021-11-15 15:15:24,100 # QDEC 0 = 0
2021-11-15 15:15:25,089 # QDEC 0 = 0
2021-11-15 15:15:26,090 # QDEC 0 = 4
2021-11-15 15:15:27,090 # QDEC 0 = 8
2021-11-15 15:15:28,090 # QDEC 0 = 20
2021-11-15 15:15:29,090 # QDEC 0 = 10
2021-11-15 15:15:30,090 # QDEC 0 = 28
2021-11-15 15:15:31,092 # QDEC 0 = 4
2021-11-15 15:15:32,090 # QDEC 0 = 36
2021-11-15 15:15:33,087 # QDEC 0 = 32
2021-11-15 15:15:34,091 # QDEC 0 = 64
2021-11-15 15:15:35,077 # QDEC 0 = 10
2021-11-15 15:15:36,081 # QDEC 0 = 46
2021-11-15 15:15:37,074 # QDEC 0 = 26
2021-11-15 15:15:38,077 # QDEC 0 = 0
2021-11-15 15:15:39,077 # QDEC 0 = 0
2021-11-15 15:15:40,069 # QDEC 0 = 0
2021-11-15 15:15:41,074 # QDEC 0 = 0
2021-11-15 15:15:42,068 # QDEC 0 = 0
2021-11-15 15:15:43,065 # QDEC 0 = 0
2021-11-15 15:15:44,066 # QDEC 0 = 0
2021-11-15 15:15:45,066 # QDEC 0 = 0
2021-11-15 15:15:45,499 # Exiting Pyterm
```

</details>

### Issues/PRs references

None

### CODE REVIEW

**Comment by benpicco** [comment]
I think you forgot to update `Makefile.ci` / Kconfig

**Comment by bergzand** [comment]
@benpicco I've added a modified version of the nrf52840dk board with qdec peripheral enabled. It is hooked up to the first two buttons and the first LED of the board. This allows for manually emulating a quadrature decoder with the buttons.

File: cpu/nrf5x_common/periph/qdec.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    gpio_init(conf(qdec)->a_pin, GPIO_IN_PU);
+    gpio_init(conf(qdec)->b_pin, GPIO_IN_PU);
+
+    dev(qdec)->PSEL.A = conf(qdec)->a_pin;
+    dev(qdec)->PSEL.B = conf(qdec)->b_pin;
+
+    /** Optionally set or disable the LED */
+    dev(qdec)->PSEL.LED = conf(qdec)->led_pin == GPIO_UNDEF ?
```
Comment (benpicco) [inline cpu/nrf5x_common/periph/qdec.c:77]:
```suggestion
    dev(qdec)->PSEL.LED = gpio_is_valid(conf(qdec)->led_pin)
```

File: cpu/nrf5x_common/periph/qdec.c [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        QDEC_PSEL_LED_CONNECT_Msk :
```
Comment (benpicco) [inline cpu/nrf5x_common/periph/qdec.c:78]:
```suggestion
      ? QDEC_PSEL_LED_CONNECT_Msk
```

File: cpu/nrf5x_common/periph/qdec.c [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        conf(qdec)->led_pin;
```
Comment (benpicco) [inline cpu/nrf5x_common/periph/qdec.c:79]:
```suggestion
      : conf(qdec)->led_pin;
```
I think it's more readable that way

File: cpu/nrf5x_common/periph/qdec.c (hunk 1 above)
Comment (bergzand) [reply cpu/nrf5x_common/periph/qdec.c]: Squashed in together with the two suggestions below

**Comment by benpicco** [comment]
looks like Kconfig is missing

**Comment by bergzand** [comment]
> looks like Kconfig is missing

Oops, amended in a kconfig file for the modded board.

**Review by benpicco — APPROVED** [review]

**Pull request merged**